delfin/drivers/hitachi/hnas/nas_handler.py

```diff
@@ -0,0 +1,825 @@
+                    quota_id = evs[0] + '-' + type + '-' + user_group_name
+                    capacity_soft_limit = \
+                        quota_map.get('Limit').replace('(Soft)', '')
+                    file_soft_limit = \
+                        quota_map.get('Limit1').replace('(Soft)', '')
+                    quota = {
+                        'native_quota_id': quota_id,
+                        'type': type,
+                        'storage_id': storage_id,
+                        'native_filesystem_id': evs[0],
+                        'native_qtree_id': qtree_id,
+                        'capacity_hard_limit': None,
+                        'capacity_soft_limit':
+                            self.get_size(capacity_soft_limit),
+                        'file_hard_limit': None,
+                        'file_soft_limit': file_soft_limit,
+                        'file_count': quota_map.get('FileCount'),
+                        'used_capacity':
+                            self.get_size(quota_map.get('Usage')),
+                        'user_group_name': user_group_name
+                    }
+                    quota_list.append(quota)
+            return quota_list
+        except exception.DelfinException as e:
+            err_msg = "Failed to get storage quota from " \
+                      "hitachi nas: %s" % (six.text_type(e))
+            LOG.error(err_msg)
+            raise e
+        except Exception as err:
+            err_msg = "Failed to get storage quota from " \
+                      "hitachi nas: %s" % (six.text_type(err))
+            LOG.error(err_msg)
+            raise exception.InvalidResults(err_msg)
+
+    def list_qtrees(self, storage_id):
+        try:
+            evs_list = self.get_fs_evs()
+            return self.get_qtree(evs_list, storage_id)
+        except exception.DelfinException as e:
+            err_msg = "Failed to get storage qtree from " \
+                      "hitachi nas: %s" % (six.text_type(e))
+            LOG.error(err_msg)
+            raise e
+        except Exception as err:
+            err_msg = "Failed to get storage qtree from " \
+                      "hitachi nas: %s" % (six.text_type(err))
+            LOG.error(err_msg)
+            raise exception.InvalidResults(err_msg)
+
+    def get_qtree(self, evs_list, storage_id):
+        qtree_list = []
+        for evs in evs_list:
+            tree_info = self.ssh_do_exec([
+                constant.CHECK_EVS % evs[1],
+                constant.TREE_INFO_COMMAND % evs[0]])
+            tree_map_list = []
+            self.split_value_map_list(tree_info,
+                                      tree_map_list,
+                                      'root',
+                                      split_key='last modified')
+            for qt_map in tree_map_list:
+                qt_name = ''
+                for key in qt_map:
+                    if qt_map[key] == '' and key != 'email':
```

Comment:
Because the parsed data treename itself is a key,value is empty, and email may also be empty. To prevent the email field from overwriting Treename, we write this
Data like the following:
tree1:
email :
root : /12323
tag : 2
usage bytes : 0 B files: 1
Last modified: the 2021-09-23 07:18:14. 714807865 + 00:00
vol2:
email :
root : /123
tag : 1
usage bytes : 0 B files: 1